src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -261,7 +261,7 @@ public void AddInterfaceImplementation(Type interfaceType)
             if (interfaceType == null)
                 throw new ArgumentNullException(nameof(interfaceType));
             if (interfaceType.IsByRef)
-                throw new ArgumentException(nameof(interfaceType));
+                throw new ArgumentException(SR.Argument_CannotGetTypeTokenForByRef);
```

Comment:
At first I updated to `throw new ArgumentException(null, nameof(interfaceType));` but the test for this is same as coreclr and failing, making the exception same as in coreclr